crates/ruff_python_formatter/src/context.rs

```diff
@@ -210,3 +220,109 @@ where
             .set_node_level(self.saved_level);
     }
 }
+
+/// The current indent level of the formatter.
+///
+/// One can determine the the width of the indent itself (in number of ASCII
+/// space characters) by multiplying the indent level by the configured indent
+/// width.
+///
+/// This is specifically used inside the docstring code formatter for
+/// implementing its "dynamic" line width mode. Namely, in the nested call to
+/// the formatter, when "dynamic" mode is enabled, the line width is set to
+/// `min(1, line_width - indent_level * indent_width)`, where `line_width` in
+/// this context is the global line width setting.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub(crate) struct IndentLevel {
+    /// The numeric level.
+    level: usize,
```

Comment:
Nit: We can probably use a `u32` here considering that we don't support files larger than 32 bytes or even `u16` because that's the most the formatter `Printer` supports.

Reply:
Yeah I was trying to figure out what type to use here. I noticed other similar types use `u32` or `u16` or whatever.

I think I do favor using fixed size types here because it provides more of a speed-bump to make sure you're careful about how they're used. On the other hand, these values do ultimately seem to get used for indexing memory, so I also find `usize` to be reasonable as well.

I'll switch this over to a `u16`. Given that it needs to do arithmetic with other `u16` values, I think that seems fine.